launch/launch/actions/execute_local.py

```diff
@@ -207,6 +209,21 @@ def __init__(
         self.__respawn = normalize_typed_substitution(respawn, bool)
         self.__respawn_delay = respawn_delay
 
+        if not respawn_max_retries:
+            # a negative value for self.__respawn_max_retries means infinite respawns
+            self.__respawn_max_retries = -1
+        else:
+            if respawn_max_retries >= 0:
+                self.__respawn_max_retries = respawn_max_retries
+            else:
+                self.__logger = launch.logging.get_logger(
```

Comment:
What would be the proper way to log this?
I used this because it was used on the same file, but if you think there is a better way please let me know.